app/src/main/java/com/example/myapplication/ui/MainActivityScreen.kt

```diff
@@ -0,0 +1,127 @@
+package com.example.myapplication.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.GridItemSpan
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Color.Companion.Gray
+import androidx.compose.ui.graphics.Color.Companion.Green
+import androidx.compose.ui.graphics.Color.Companion.Magenta
+import androidx.compose.ui.graphics.Color.Companion.Red
+import androidx.compose.ui.graphics.Color.Companion.Yellow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import kotlin.random.Random.Default.nextInt
+
+sealed interface ItemsHeight {
+    val heightPercent: Int
+
+    data class Fixed(override val heightPercent: Int, val height: Int) : ItemsHeight
+    data class Dynamic(override val heightPercent: Int) : ItemsHeight
+}
+
+data class ItemViewState(
+    val title: String,
+    val background: Color,
+)
+
+data class GridViewState(
+    val items: List<ItemViewState>,
+    val gridBackground: Color,
+    val columnsCount: Int,
+    val itemsHeight: ItemsHeight,
+)
```

Comment:
ViewStates better moved into a separate file because they will use in ViewModels and Screens file